File: src/test/scala/com/exasol/cloudetl/helper/ExportParallelismCalculatorTest.scala
package com.exasol.cloudetl.helper

import com.exasol.ExaMetadata
import com.exasol.cloudetl.storage.StorageProperties

import org.mockito.Mockito._
import org.scalatest.BeforeAndAfterEach
import org.scalatest.funsuite.AnyFunSuite
import org.scalatestplus.mockito.MockitoSugar

class ExportParallelismCalculatorTest extends AnyFunSuite with BeforeAndAfterEach with MockitoSugar {
  private val metadata: ExaMetadata = mock[ExaMetadata]

  override def beforeEach(): Unit = {
    when(metadata.getNodeCount()).thenReturn(1)
    when(metadata.getMemoryLimit()).thenReturn(new java.math.BigInteger("2000000000")) // 2GB
    ()
  }

  test("getParallelism() returns user provided value") {
    val properties = StorageProperties(Map("PARALLELISM" -> "iproc()"))
    assert(ExportParallelismCalculator(metadata, properties).getParallelism() === "iproc()")
  }

  test("getParallelism() returns calculated value") {
    assert(
      ExportParallelismCalculator(metadata, StorageProperties(Map.empty[String, String]))
        .getParallelism()
        .contains("iproc(), mod(rownum,")
    )
  }

}

File: src/test/java/com/exasol/cloudetl/helper/ExportParallelismCalculatorTest.java
package com.exasol.cloudetl.helper;

import static org.junit.jupiter.api.Assertions.*;
import static org.mockito.Mockito.*;

import java.math.BigInteger;
import java.util.Map;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import com.exasol.ExaMetadata;
import com.exasol.cloudetl.storage.StorageProperties;

class ExportParallelismCalculatorTest {
    private final ExaMetadata metadata = mock(ExaMetadata.class);

    @BeforeEach
    void beforeEach() {
        when(this.metadata.getNodeCount()).thenReturn(1L);
        when(this.metadata.getMemoryLimit()).thenReturn(new BigInteger("2000000000"));
    }

    @Test
    void getParallelismReturnsUserProvidedValue() {
        final StorageProperties properties = new StorageProperties(Map.of("PARALLELISM", "iproc()"));
        assertEquals("iproc()", new ExportParallelismCalculator(this.metadata, properties).getParallelism());
    }

    @Test
    void getParallelismReturnsCalculatedValue() {
        assertTrue(new ExportParallelismCalculator(this.metadata, new StorageProperties(Map.of())).getParallelism()
                .contains("iproc(), mod(rownum,"));
    }
}
